### PR TITLE
unicode: apply unicode normalization

### PR DESCRIPTION
**Pull request by missinglink**
as per https://github.com/pelias/api/pull/1373 and https://github.com/pelias/api/issues/1375 this PR uses the same unicode normalization form (NFKC) as used in [pelias/api](https://github.com/pelias/api/blob/master/helper/unicode.js).

it's a relatively simple change but will improve matching in general, and ensure that the index is generated with strings in the same unicode form as we provide at query time.

note: the original description is a little misleading, there doesn't need to be a match in unicode normalisation forms between any other system and `placeholder`. I originally thought that this was the case but in hindsight it's better to have this lib accept any form of normalisation and for it to 'just work'. In order to accomplish this we provide an internal normalisation function which is applied *both* at indexing time *and* query time.

### CODE REVIEW

**Comment by missinglink** [comment]
There were a fair few changes, I've uploaded a copy here https://gist.github.com/missinglink/36eca310d6bb811c7aeda99a32d166e2 (8MB)

> note: `<` refers to the `master` branch and `>` refers to the new `unicode` branch

It doesn't cause any test failures, but I would like to consider this change a little more before merging.

**Comment by missinglink** [comment]
opening up for review/merging, I would like to see `placeholder` have symmetry in the unicode normalization forms at index vs. search time.

one issue will be that when we release this code the search behaviour of new code+old index will change, this will resolve itself once the new code is paired with a new index but there will be a transitionary period, I may end up just marking it as a breaking change.

File: prototype/query.js [hunk 1]
```diff
@@ -2,7 +2,6 @@
 var async = require('async');
 var util = require('util');
 var Result = require('../lib/Result');
-var sorted = require('../lib/sorted');
```
Comment (missinglink) [inline prototype/query.js:5]: cleanup: unused function

File: prototype/tokenize.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
 
 // plugin for tokenize
-const _ = require('lodash'),
-    async = require('async'),
-    analysis = require('../lib/analysis'),
-    permutations = require('../lib/permutations');
+const _ = require('lodash');
+const async = require('async');
+const analysis = require('../lib/analysis');
+const permutations = require('../lib/permutations');
```
Comment (missinglink) [inline prototype/tokenize.js:6]: cleanup: modern syntax

**Comment by missinglink** [comment]
this PR includes a [commit to fix the CI](https://github.com/pelias/placeholder/pull/207/commits/a66655f8e77db7d7b27b0b6ac790cfb979e9977d).

the CI script historically contained the `archive` path prefix although that directory doesn't exist in either S3 or on the Bunny CDN 🤷‍♂️

**Comment by missinglink** [comment]
right, so this is good-to-go, it's a much needed and positive change, but...

we really want to ensure that the code and the data match, there's a small window of potential incompatibility where matching data produced *after* merging, with code from *before*, will result in some minor regressions.

I think the best course of actions is:
- merge the code
- immediately publish the new data at the default download location

For developers running Pelias they shoud:
- update the code/image and the data and re-create the `placeholder` docker containers

**Comment by orangejulius** [comment]
I've updated our public Placeholder snapshot with a build based on this branch, so all the tests now pass.

We've actually discovered that there appear to be minimal issues if versions of Placeholder prior to this branch are used with data built using the new normalization from this branch, so it's safe to merge!

**Pull request merged**